tenseal/tensors/ckksvector.cpp

```diff
@@ -200,4 +201,44 @@ CKKSVector& CKKSVector::mul_plain_inplace(vector<double> to_mul) {
     return *this;
 }
 
+CKKSVector CKKSVector::matmul_right(const vector<vector<double>> matrix) {
+    // matrix is organized by rows
+    // _check_matrix(matrix, this->size())
+    size_t n_rows = matrix.size();
+    size_t n_cols = matrix[0].size();
+
+    if (this->size() != matrix.size()) {
+        cout << this->size() << " vs " << n_rows << endl;
+        throw invalid_argument("matrix shape doesn't match with vector size");
+    }
+
+    CKKSVector result_vector = *this;
```

Comment:
This is a bit weird. If in the future anyone adds an internal pointer member in the CKKSVector, it can create issues if it not updated everywhere.

Maybe a new CKKSVector for the add_many output is better.
And a method CKKSVector::Reload/Update, to update the internal state in a single place, if it's needed for the current context too.

Reply:
This got updated, let me know what you think about the new matmul function regarding this, `CKKSVector new_vector = *this;` is mainly doing a copy, isn't it the proper way of doing?